.circleci/config.yml

```diff
@@ -471,40 +488,21 @@ jobs:
       CC: clang
       CXX: clang++
       MAKEFLAGS: -j 10
-    steps:
-      - checkout
-      - run: *run_build
-      - store_artifacts: *artifacts_solc
-      - persist_to_workspace: *artifacts_executables
+    <<: *steps_b_ubu
 
   b_ubu_asan_clang: &b_ubu_asan_clang
-    docker:
-      - image: << pipeline.parameters.ubuntu-2004-clang-docker-image >>
+    <<: *b_ubu_clang
     environment:
-      CC: clang
-      CXX: clang++
-      CMAKE_OPTIONS: -DSANITIZE=address
       MAKEFLAGS: -j 3
-    steps:
-      - checkout
-      - run: *run_build
-      - store_artifacts: *artifacts_solc
-      - persist_to_workspace: *artifacts_executables
+      CMAKE_OPTIONS: -DSANITIZE=address
 
   b_ubu_ubsan_clang: &b_ubu_ubsan_clang
-    docker:
-      - image: << pipeline.parameters.ubuntu-2004-clang-docker-image >>
+    <<: *b_ubu_clang
     environment:
-      CC: clang
-      CXX: clang++
-      CMAKE_OPTIONS: -DSANITIZE=undefined
       MAKEFLAGS: -j 3
+      CMAKE_OPTIONS: -DSANITIZE=undefined
     steps:
-      - checkout
-      - run: *run_build
       - run: *gitter_notify_failure
-      - store_artifacts: *artifacts_solc
-      - persist_to_workspace: *artifacts_executables
```

Comment:
`steps` will overwrite `steps` from `b_ubu_clang` as well. You need the trick with adding nested steps using `when`.